### PR TITLE
#52021 - added deletion of uploaded file. Redesigned file upload button.

### PR DESCRIPTION
**Pull request by ManuBoca92**

### CODE REVIEW

File: examples/now/pages/index.vue [hunk 1]
```diff
@@ -66,9 +66,14 @@
               </b-upload>
               <span
                 v-if="file"
```
Comment (bartdominiak) [inline examples/now/pages/index.vue:69]: I think we should hide "Click to upload" when user provide file (if it support only one file attachment)

```vue
...
<b-upload
  v-if="!file"
  :name="field.name"
  v-model="file"
  :required="field.required"
>
  <a class="tag is-secondary">
    <b-icon icon="upload" />
    <span>
      Click to upload
    </span>
  </a>
</b-upload>
<span
  v-else
  class="tag is-primary"
>
  {{ file.name }}
  <button
    @click="deleteFile()"
    class="delete is-small"
    type="button"
  />
</span>
...
```

File: examples/now/pages/index.vue [hunk 2]
```diff
@@ -135,6 +140,9 @@ export default {
     isCheckbox(type) {
       return type === 'checkbox'
     },
+    deleteFile() {
+      this.file = null
+    },
```
Comment (bartdominiak) [inline examples/now/pages/index.vue]: IMO it's unecessary to add another method, you can just add
```vue
<button @click="file = null" />

Comment (ManuBoca92) [reply examples/now/pages/index.vue]: Done.

File: examples/now/pages/index.vue (hunk 1 above)
Comment (ManuBoca92) [reply examples/now/pages/index.vue:69]: I used `v-show="!file"` instead since `v-if="!file"` prevents uploaded file from being sent.

Comment (bartdominiak) [reply examples/now/pages/index.vue]: After that you can add `v-else` instead of `v-if="file"`

Comment (bartdominiak) [reply examples/now/pages/index.vue:69]: Yeah, You're right. You need to use v-show to keep this `input` value in form.

Keep this in mind. In Javascript we have:
```javascript
const test = null
test === true // return false
test === false // return false
!test === true // return true
!test === false // return false
```
So, `v-if="file"` will return true only if file is added. If it's `null` it wont render anything

**Pull request merged**